execution_workflows/DaPars/bin/check_bedgraph.py

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env python3
+
+import sys
+import argparse
+
+def parse_args(args=None):
+	Description = "Reformat DaPars bedgraph input to have leading chr for all sequence regions."
+	Epilog = "Example usage: python check_bedgraph.py <FILE_IN> <FILE_OUT>"
+
+	parser = argparse.ArgumentParser(description=Description, epilog=Epilog)
+	parser.add_argument("FILE_IN", help="Input bedgraph file.")
+	parser.add_argument("FILE_OUT", help="Output bedgraph file.")
+	return parser.parse_args(args)
+
+def check_bedgraph(file_in, file_out):
+	"""
+	This function checks that there is leading chr for all sequence regions
+	Otherwise, add the leading chr
```

Comment:
@ninsch3000 The author Zheng Xia got back to me confirming that the tool requires the “chr” prefix for input and that we need to add leading chr to all files

I’ll look if there is a way to convert ensembl to gencode properly but if you have any ideas, let me know

Reply:
Sorry this is a little late, hope you haven't spent too much time researching this. I think we decided in the 'utils' meeting last Friday to use Gencode & standard chromosomes as the default annotation file for all execution workflows, so hopefully the 'chr' prefix will be propagated through gtf --> bed12 conversions. I'm comfortable with the prefix check remaining in this case as the tool hardcodes for it.

Reply:
Hi @faricazjj and @SamBryce-Smith , thanks for checking this out. I agree, the prefix check should definitely be there if the tool hardcodes for it. And yes, we will have the chr prefix in our annotations in the future. So no need to worry for now.

Reply:
@ninsch3000 @SamBryce-Smith Sounds good, i'll add a prefix check here instead! Just to avoid errors later in the pipeline, should we remove rows without the "chr" prefix in this check? 

Also, I missed the meeting yesterday. Do we have an update on when the new gtf file with "chr" prefix is going to be added in the test_data folder to test this workflow with once I've updated this file?

Reply:
Personally I'd rather be cautious and throw an error if rows do not have the chr prefix. At least for our challenge, the BAM files should have been generated using Gencode scaffolds & annotations, so the chr prefix should be propagated throughout the steps. We're also going to subset to canonical chromosomes so they all should have the prefix anyway. We should definitely note in the README that the tool requires the `chr` prefix so input files should only contain chromosomes named as such.

@ninsch3000 thoughts on how strict to be here?

Have messaged @mrgazzara  about how exactly to maintain the folder. Am hoping to get new test data merged in the next couple of days. I'll ping the Slack when it's done.

Reply:
Agree with @SamBryce-Smith , throwing an error (with hint that a different gtf file should be used, or "chr" should be added to all entries at the user's own risk) would be my choice as well. And mentioning the hardcoded "chr" in the README, too. But given the fact that we want to use Gencode with "chr" anyway, as Sam says, for our challenge all should be fine and we shouldn't spent too much time on fixing what is definitely the tool's flaw, not ours ;) That said, I'm still a fan of documentation and error checks.
 
Also agree that we should get some proper test files with "chr" soon, as we should also test all other tools with those. Let's see what @mrgazzara thinks!

Reply:
Thanks guys @SamBryce-Smith @ninsch3000 ! I updated the readme under the notes section to mention the error thrown if a sample file has a row without leading 'chr', including the screenshot of the error message shown when this happens :D